### PR TITLE
Improved logic in the receiver tactic and added tests for the receiver position

### PR DESCRIPTION
**Pull request by MathewMacDougall**
### Description

Improved the logic for the receiver tactic to make it more robust, especially with angle calculations.

Also added a parameterized test to make sure the receiver is always trying to go to the right location. This test was added because during field testing we observed the robot would sometimes move away from where we expected, in what looked like a logic error rather than bad vision.

### Testing Done

All old unit tests pass
New tests added and all pass

### Resolved Issues

None

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom`
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by garethellis0 — DISMISSED** [review]
LGTM :+1: :tada: 

I field-tested the rougher version of this as well. CI failures here are unrelated, having some issues with coroutines that should hopefully be resolved soon.

File: src/thunderbots/software/ai/hl/stp/tactic/receiver_tactic.cpp [hunk 1]
```diff
@@ -171,8 +139,12 @@ Angle ReceiverTactic::getOneTimeShotDirection(const Ray& shot, const Ball& ball)
     Vector shot_vector = shot.getDirection();
     Angle shot_dir     = shot.getDirection().orientation();
 
-    Point ball_vel       = ball.velocity();
-    Point lateral_vel    = ball_vel.project(shot_vector.norm().perp());
+    Point ball_vel    = ball.velocity();
```
Comment (EvanMorcom) [inline src/thunderbots/software/ai/hl/stp/tactic/receiver_tactic.cpp]: Nit: 

Usually, the convention is to make directions 'Vector's

**Review by EvanMorcom — DISMISSED** [review]
LGTM :+1: 

Just one nit.

File: src/thunderbots/software/ai/hl/stp/tactic/receiver_tactic.cpp (hunk 1 above)
Comment (MathewMacDougall) [reply src/thunderbots/software/ai/hl/stp/tactic/receiver_tactic.cpp]: Yup, you're right. Good catch. Fixed.

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada:

**Pull request merged**